### PR TITLE
Feat/history fetch till completion

### PR DESCRIPTION
**Pull request by devceline**
# Changes
- Instead of fetching 200 messages, fetch in a loop until there are no more records

### CODE REVIEW

File: packages/notify-client/src/utils/history.ts [hunk 1]
```diff
@@ -69,18 +71,39 @@ export const fetchAndInjectHistory = async (
...
+        : baseFetchParams;
+
+      const messages = await historyClient.getMessages(fetchParams);
+
+      core.logger.info(
+        `Fetched ${messages.messageResponse.messages.length} messages from history for topic: ${topic}, store: ${name}`
+      );
+
+      retrievedCount = messages.messageResponse.messages.length;
+      originId =
```
Comment (bkrem) [inline packages/notify-client/src/utils/history.ts:99]: Could you add context on what is being done with `originId` please? Presumably being used as a cursor

File: packages/notify-client/src/utils/history.ts [hunk 2]
```diff
@@ -69,18 +71,39 @@ export const fetchAndInjectHistory = async (
...
+        messages.messageResponse.messages[
+          messages.messageResponse.messages.length - 1
+        ].message_id;
 
-    const currentMessages = messages.messageResponse.messages;
-    await reduceAndInjectHistory(core, currentMessages, topic);
+      const currentMessages = messages.messageResponse.messages;
+      await reduceAndInjectHistory(core, currentMessages, topic);
+    } while (retrievedCount === HISTORY_MAX_FETCH_SIZE);
```
Comment (bkrem) [inline packages/notify-client/src/utils/history.ts:106]: Open `do/while` loop is making me a bit nervous for infinite loop possibilities 🤔 

Is there a scenario where `retrievedCount` will be set to `HISTORY_MAX_FETCH_SIZE` but then never update, e.g. if further fetching fails silently?

**Review by bkrem — APPROVED** [review]
Approved but have some concerns with the trappings of `do/while` loops

File: packages/notify-client/src/utils/history.ts [hunk 3]
```diff
@@ -69,21 +71,44 @@ export const fetchAndInjectHistory = async (
...
+
+        const currentMessages = messages.messageResponse.messages;
+        await reduceAndInjectHistory(core, currentMessages, topic);
+      } catch (e) {
+        retrievedCount = 0;
+        core.logger.error(`Failed to fetch and inject history for ${name}`, e);
+        break;
+      }
+    } while (retrievedCount === HISTORY_MAX_FETCH_SIZE);
+  } catch (e: any) {}
```
Comment (chris13524) [inline packages/notify-client/src/utils/history.ts:113]: Why no handling? We won't know if something bad happens

**Review by chris13524 — APPROVED** [review]

**Pull request merged**